nostr/save_nostr_key.js

```diff
@@ -0,0 +1,82 @@
+const asyncAuto = require('async/auto');
+const {returnResult} = require('asyncjs-util');
+
+const {encryptToNode} = require('../encryption');
+const {homePath} = require('../storage');
+
+const nostrKeyFile = 'nostr_private_key';
+
+/** Save encrypted nostr private key
+
+  {
+    fs: {
+      getFile: <Read File Contents Function> (path, cbk) => {}
+      makeDirectory: <Make Directory Function> (path, cbk) => {}
+      writeFile: <Write File Contents Function> (path, contents, cbk) => {}
+    }
+    key: <Nostr Private Key String>
+    lnd: <Authenticated LND API Object>
+    logger: <Winston Logger Object>
+  }
+
+  @returns via cbk or Promise
+*/
+module.exports = (args, cbk) => {
+  return new Promise((resolve, reject) => {
+    return asyncAuto({
+      // Check arguments
+      validate: cbk => {
+        if (!args.fs) {
+          return cbk([400, 'ExpectedFilesystemMethodsToSaveNostrKey']);
+        }
+
+        if (!args.key) {
+          return cbk([400, 'ExpectedNostrPrivateKeyToSaveNostrKey']);
+        }
+
+        if (!args.lnd) {
+          return cbk([400, 'ExpectedLndToSaveNostrKey']);
+        }
+
+        if (!args.logger) {
+          return cbk([400, 'ExpectedLoggerToSaveNostrKey']);
+        }
+
+        return cbk();
+      },
+
+      // Register the home directory
+      registerHomeDir: ['validate', ({}, cbk) => {
+        return args.fs.makeDirectory(homePath({}).path, err => {
+          // Ignore errors, the directory may already be there
+          return cbk();
+        });
+      }],
+
+      // Encrypt the private key
+      encryptKey: ['validate', ({}, cbk) => {
+        return encryptToNode({
+          lnd: args.lnd,
+          message: args.key,
+        },
+        cbk)
+      }],
+
+      // Save the encrypted nostr private key
+      saveKey: ['encryptKey', 'registerHomeDir', ({encryptKey}, cbk) => {
+        const {path} = homePath({file: nostrKeyFile});
```

Comment:
i think this will break because you can potentially have multiple different nodes, so if you use the wrong node with this then it won't decrypt

how about one nostr.json file that has all the nostr things in it, and then that file can include an array of nostr identities?